src/uwtools/tests/config/test_tools.py

```diff
@@ -374,6 +364,80 @@ def test_realize_config_simple_yaml(tmp_path):
     help_realize_config_simple("simple2.yaml", FORMAT.yaml, tmp_path)
 
 
+def test_realize_config_single_dereference(caplog, tmp_path):
+    path = tmp_path / "a.yaml"
+    supplemental_path = tmp_path / "b.yaml"
+    with writable(path) as f:
+        yaml.dump({"1": "a", "2": "{{ deref }}", "3": "{{ temporalis }}"}, f)
+    with writable(supplemental_path) as f2:
+        yaml.dump({"2": "b", "temporalis": "c"}, f2)
+    tools.realize_config(
+        input_config=path,
+        input_format=FORMAT.yaml,
+        output_file=None,
+        output_format=FORMAT.yaml,
+        supplemental_configs=[supplemental_path],
+        dry_run=True,
+    )
+    expected = """'1': a
+'2': b
+'3': c
+temporalis: c
+"""
+    actual = "\n".join(record.message for record in caplog.records)
+    assert actual == expected
+
+
+def test_realize_config_supp_list(caplog, tmp_path):
+    path = tmp_path / "a.yaml"
+    supplemental_path = tmp_path / "b.yaml"
+    second_supp_path = tmp_path / "c.yaml"
+    with writable(path) as f:
+        yaml.dump({"1": "a", "2": "{{ deref }}", "3": "{{ temporalis }}"}, f)
+    with writable(supplemental_path) as f2:
+        yaml.dump({"2": "b", "temporalis": "c"}, f2)
+    with writable(second_supp_path) as f3:
+        yaml.dump({"4": "d", "tempus": "fugit", "deref": "{{ tempus }}"}, f3)
+    tools.realize_config(
+        input_config=path,
+        input_format=FORMAT.yaml,
+        output_file=None,
+        output_format=FORMAT.yaml,
+        supplemental_configs=[supplemental_path, second_supp_path],
+        dry_run=True,
+    )
+    expected = """'1': a
+'2': b
+'3': c
+'4': d
+deref: fugit
+temporalis: c
+tempus: fugit
+"""
+    actual = "\n".join(record.message for record in caplog.records)
+    assert actual == expected
+
+
+def test_realize_config_supp_none(caplog, tmp_path):
+    path = tmp_path / "a.yaml"
+    with writable(path) as f:
+        yaml.dump({"1": "a", "2": "{{ deref }}", "3": "{{ temporalis }}"}, f)
+    tools.realize_config(
+        input_config=path,
+        input_format=FORMAT.yaml,
+        output_file=None,
+        output_format=FORMAT.yaml,
+        supplemental_configs=None,
+        dry_run=True,
+    )
+    expected = """'1': a
+'2': '{{ deref }}'
+'3': '{{ temporalis }}'
+"""
+    actual = "\n".join(record.message for record in caplog.records)
+    assert actual == expected
```

Comment:
Great demonstrations of none/one/many correctness in these tests. A couple ideas for your consideration -- not asking for a change, just in case you find any of it helpful.

- You could omit the `=None` arguments, as they're IIRC all optional and supply `None` values themselves. Maybe there's a benefit to being explicit, but you could make these tests slightly more concise that way.

- You could reuse the variable `f` (instead of doing `f2` etc.) when doing the various file writes, because the `with` context manager closes the file beyond the block such that `f` has no useful value after the block:
```
>>> with open("/tmp/foo", "w") as f:
...   print("bar", file=f)
... 
>>> print("baz", file=f)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: I/O operation on closed file.
```
I think it's idiomatic to keep it simple and just use `f` repeatedly.

- You could use `strip()` to e.g.
```
>>> s = """
... hello
... world
... """.strip()
>>> print(s)
hello
world
```
IMO it's nice to see the `expected` output lined up nicely, without the first line of output being at the end of a line of code.

- Instead of `dry_run=True`, you could leave it blank and the output will appear on `stdout` (since you have `output_file=None` -- which you could also omit, per the first bullet point above), which you can read from `capsys` instead of `caplog`. There are examples of this in the test you can borrow from. This is probably actually safer, because technically we could decide to also log other messages alongside the ones showing the realized config, and then these tests would unexpectedly fail. But when we write the output to `stdout` (as opposed to logging to `stderr`), it will be guaranteed to be exactly the same as what would be written to a file -- i.e. it's downstream-programmatically-consumable text, not human-readable text -- and so safe to make an assertion of equality about.

Reply:
I changed the `f2`s and `f3`s to `f`s, removed a bunch of the `=None` lines, and replaced `caplog` with `capsys`, but was only able to get `.strip()` working when I applied it to both `expected` and `actual`. In order to avoid having it both places, I left the lines as they were. I do appreciate all of the suggestions.